apps/src/applab/ImportScreensDialog.jsx

```diff
@@ -249,7 +261,77 @@ var ImportScreensDialog = React.createClass({
   }
 });
 
-export default ImportScreensDialog;
+/**
+ * Helper function that takes a dom node and returns an object that conforms
+ * to an importableScreenShape, checking for assets, and conflicting ids.
+ */
+function getImportableScreen(dom) {
+  const id = dom.id;
+  const willReplace = designMode.getAllScreenIds().includes(id);
+  const conflictingIds = [];
+  Array.from(dom.children).forEach(child => {
+    if (!elementUtils.isIdAvailable(child.id)) {
+      var existingElement = elementUtils.getPrefixedElementById(child.id);
+      if (existingElement && elementUtils.getId(existingElement.parentNode) !== id) {
+        conflictingIds.push(child.id);
+      }
+    }
+  });
+
+  // TODO: filter out assets that will just be imported without replacing anything.
```

Comment:
I could either do this as part of this pull request, or a future pull request. The feature is hidden behind an experiment so nobody will see it in a half-implemented state.